src/parser.js

```diff
@@ -16,7 +16,12 @@
  */
 const parseAuthFailedMessage = (message) => {
     const regex = /\w+ \w+ \w+ ?\w+ ?\w+ (?<username>\w+) \w+ (?<ip>.*) \w+ (?<port>\w+) \w+/;
-    return regex.exec(message).groups;
+    const parsed = regex.exec(message);
```

Comment:
We can use the optional chaining here to return `regex.exec(message)?.groups` instead.

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
nice :-)